scripts/release/prepare.sh

```
@@ -0,0 +1,289 @@
+    if [[ "$DRY_RUN" == "true" ]]; then
+        log_info "DRY RUN: Skipping repository validation and git sync"
+        return 0
+    fi
+    validate_repository
+    git fetch upstream
+    git checkout main
+    git pull --ff-only upstream main
+}
+
+validate_environment() {
+    if [[ "$DRY_RUN" == "true" ]]; then
+        log_info "DRY RUN: Skipping GitHub CLI validation"
+        return 0
+    fi
+    validate_gh_cli
+}
+
+
+determine_current_versions() {
+    read -r current_v1 current_v2 <<< "$(get_current_versions)"
+    log_info "Current versions: $current_v1 / $current_v2"
+    if [[ -z "${current_v1:-}" || -z "${current_v2:-}" ]]; then
+        log_error "Failed to determine current versions (v1='${current_v1:-}', v2='${current_v2:-}')."
+        log_error "Ensure 'get_current_versions' returns both versions in 'vX.Y.Z vA.B.C' format."
+        exit 1
+    fi
+}
+
+
+validate_semver_increment() {
+    local new_version="$1"
+    local current_version="$2"
+    # Expect format vX.Y.Z
+    if ! [[ "$new_version" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+        log_error "Version $new_version does not follow semantic versioning format (vX.Y.Z)"
+        return 1
+    fi
+    # No monotonicity check; start.sh is the source of truth
+    return 0
+}
+
+normalize_and_validate_versions() {
+    new_version_v1=$(echo "$new_version_v1" | sed 's/^v//')
+    new_version_v1="v$new_version_v1"
+    new_version_v2=$(echo "$new_version_v2" | sed 's/^v//')
+    new_version_v2="v$new_version_v2"
+    validate_semver_increment "$new_version_v1" "$current_v1" || exit 1
+    validate_semver_increment "$new_version_v2" "$current_v2" || exit 1
+}
+
+generate_changelog() {
+    log_info "Generating changelog..."
+    temp_changelog=$(mktemp)
+    trap "rm -f \"$temp_changelog\"" EXIT
+    if make changelog > "$temp_changelog" 2>/dev/null; then
+        log_success "Changelog generated successfully"
+    else
+        log_warning "make changelog failed, using fallback template"
```

fail hard, no fallback